### PR TITLE
Processing refactor

### PR DESCRIPTION
**Pull request by kingman**
re-organize the folder structure of components/processing to adhere agreed upon structure of 
components/processing/
── libs
── terraform

### CODE REVIEW

File: components/processing/README.md [hunk 1]
```diff
@@ -36,5 +36,5 @@ Running invoke.sh will do the following -
 
 ### Deployments
 There are Python-based deployments, such as in [deployments/cloud_function/main.py](deployments/cloud_function/main.py) which leverage one or more libraries. There is an associates requirements.in and requirements.txt for these libraries (using -r) capturing the full non-library dependencies. It is assumed that all of the relevant libraries will be staged in the deployment.
-The instructions for Cloud Function deployment is available in the [README.md](deployments/cloud_function/README.md) and for Cloud Run is available in the [README.md](deployments/cloud_run/README.md).
+The instructions for Cloud Function deployment is available in the [README.md](deployments/cloud_function/README.md) and for Cloud Run is available in the [README.md](terraform/README.md).
```
Comment (eyalbenivri) [inline components/processing/README.md:39]: Do we still have this support deployment to cloud_function? should we get rid completely of the `deployments` folder?

File: components/processing/terraform/build/requirements.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # This file was autogenerated by uv via the following command:
-#    uv pip compile --generate-hashes -c /usr/local/google/home/scannell/github/document-processing-and-understanding/reqs/constraints.txt /usr/local/google/home/scannell/github/document-processing-and-understanding/components/processing/deployments/cloud_run/build/pyproject.toml
+#    uv pip compile --generate-hashes -c /usr/local/google/home/chwa/github/document-processing-and-understanding/reqs/constraints.txt /usr/local/google/home/chwa/github/document-processing-and-understanding/components/processing/terraform/build/pyproject.toml
```
Comment (eyalbenivri) [inline components/processing/terraform/build/requirements.txt:2]: if this file is containing paths from your working machine, then we probably shouldn't have it tracked by git?

File: components/processing/terraform/main.tf [hunk 3]
```diff
@@ -98,7 +101,7 @@ module "gcloud" {
   version = "~> 3.4"
 
   create_cmd_entrypoint = "gcloud"
-  create_cmd_body       = "builds submit --region ${var.region} --project ${var.project_id} --config \"${local_file.cloudbuild_cloud_run.filename}\" \"${path.module}/../../../..\""
+  create_cmd_body       = "builds submit --region ${var.region} --project ${var.project_id} --config \"${local_file.cloudbuild_cloud_run.filename}\" \"${path.module}/../../..\""
```
Comment (eyalbenivri) [inline components/processing/terraform/main.tf:104]: I believe Anurhada's request was to separate the build/deployment related "resources" to a different file (like `build.tf`) so it's easier to users to remove that in favor of a proper CI/CD pipeline.

File: components/processing/README.md (hunk 1 above)
Comment (dharmez) [reply components/processing/README.md:39]: My vote is to remove the deployment of the customer document processor to cloud_function. Let's just use Cloud Run. If a customer or partner wants to deploy it on cloud function or GKe.. they can do it themselves.

Comment (kingman) [reply components/processing/README.md]: `deployments` folder removed

File: components/processing/terraform/main.tf (hunk 3 above)
Comment (kingman) [reply components/processing/terraform/main.tf]: break out cloud build yaml and cloud build resource into `build.tf`

File: components/processing/terraform/build.tf [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    registry_url                  = local.registry_url,
+    region                        = var.region,
+    service_account               = module.doc_processor_account.email
+    processing_cloud_run_job_name = var.processing_cloud_run_job_name
+    build_service_account         = var.cloud_build_service_account_email
+  })
+}
+
+# See https://github.com/terraform-google-modules/terraform-google-gcloud
+module "gcloud" {
```
Comment (eyalbenivri) [inline components/processing/terraform/build.tf:42]: Didn't we wanted to move this into a separate `build.tf`` file?

**Review by eyalbenivri — APPROVED** [review]

**Pull request merged**